cnd/src/swap_protocols/facade.rs

```diff
@@ -81,8 +81,8 @@ impl Facade {
         &self,
         id: LocalSwapId,
         swap_params: HanEtherereumHalightBitcoinCreateSwapParams,
-    ) {
-        self.swarm.initiate_communication(id, swap_params).await;
+    ) -> anyhow::Result<()> {
+        self.swarm.initiate_communication(id, swap_params).await
```

Comment:
It is a question of style but I prefer to never return `Result`s directly because I think it is less readable of what is going on.
```suggestion
        self.swarm.initiate_communication(id, swap_params).await?;
        
        Ok(())
```

Reply:
I agree with you @thomaseizinger for a longer function but on a one line function like this I think that the added `?` and `Ok(())` are just noise.